dev-libs/mpc/mpc-1.1.0.recipe

```diff
@@ -0,0 +1,70 @@
+SUMMARY="Arithmetics of complex numbers with arbitrarily high precision"
+DESCRIPTION="GNU Mpc is a C library for the arithmetic of complex numbers \
+with arbitrarily high precision and correct rounding of the result. It \
+extends the principles of the IEEE-754 standard for fixed precision real \
+floating point numbers to complex numbers, providing well-defined semantics \
+for every operation. At the same time, speed of operation at high precision \
+is a major design goal."
+HOMEPAGE="http://www.multiprecision.org/mpc/"
+COPYRIGHT="2006-2011 INRIA"
+LICENSE="GNU LGPL v3"
+REVISION="1"
+SOURCE_URI="https://ftpmirror.gnu.org/mpc/mpc-$portVersion.tar.gz"
+CHECKSUM_SHA256="6985c538143c1208dcb1ac42cedad6ff52e267b47e5f970183a3e75125b43c2e"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion=3.1.0
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	mpc$secondaryArchSuffix = $portVersion
+	lib:libmpc$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libgmp$secondaryArchSuffix
+	lib:libmpfr$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	mpc${secondaryArchSuffix}_devel = $portVersion
+	devel:libmpc$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	mpc$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libgmp$secondaryArchSuffix
+	devel:libmpfr$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:awk
+	cmd:gcc$secondaryArchSuffix
+	cmd:make
+	"
+
+BUILD()
+{
+	runConfigure ./configure
+	make $jobArgs
+}
+
+defineDebugInfoPackage mpc$secondaryArchSuffix \
+	$libDir/libmpc.so.$libVersion
+
+INSTALL()
+{
+	make install
+	rm $libDir/libmpc.la
+	prepareInstalledDevelLib libmpc
+	packageEntries devel $developDir
+}
+
+TEST()
+{
+	make $jobArgs check
```

Comment:
Shall we drop the `documentation/info/mpc.info` on secondary architectures and disable parallel builds in `TEST()`?
```diff
 INSTALL()
 {
 	make install
 	rm $libDir/libmpc.la
+	if [ -n "$secondaryArchSuffix" ]; then
+		rm -rf $infoDir
+	fi
 	prepareInstalledDevelLib libmpc
 	packageEntries devel $developDir
 }
 
 TEST()
 {
-	make $jobArgs check
+	make check
 }
```

Everything else looks OK. Built on x86_gcc2, x86 second arch and x86_64.
Tested on x86 secondary arch and x86_64. All 67 tests pass on these architectures.

On x86_gcc2, however, gcc2 cannot build `make check`.

Reply:
Oops, I just notice this line:
```
/boot/system/develop/headers/posix/signal.h:75: warning: ANSI C doesn't support unnamed structs/unions
```
when I run `hp --test mpc` on x86_gcc2:
```
In file included from /boot/system/develop/headers/bsd/signal.h:9,
                 from /boot/system/develop/headers/posix/sys/wait.h:10,
                 from /boot/system/develop/headers/bsd/sys/wait.h:9,
                 from /boot/system/develop/headers/posix/stdlib.h:14,
                 from /boot/system/develop/headers/bsd/stdlib.h:9,
                 from /sources/mpc-1.1.0/tests/random.c:26:
/boot/system/develop/headers/posix/signal.h:75: warning: ANSI C doesn't support unnamed structs/unions
/sources/mpc-1.1.0/tests/random.c: In function `test_random_d':
/sources/mpc-1.1.0/tests/random.c:234: initializer element is not computable at load time
/sources/mpc-1.1.0/tests/random.c:234: (near initialization for `mpfr_x[0]')
make[2]: *** [Makefile:1303: random.lo] Error 1
make[2]: Leaving directory '/sources/mpc-1.1.0/tests'
```
Shouldn't the [headers/posix/signal.h](https://github.com/haiku/haiku/blob/master/headers/posix/signal.h#L75) shipped in x86_gcc2 be gcc2-compatible?

Reply:
@fbrosson It's compatible in the sense that gcc2 doesn't panic upon meeting. I think the warning could be easily silenced. You should file a ticket on Trac for this.